x-pack/plugins/observability_solution/observability_onboarding/server/routes/kubernetes/route.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { v4 as uuidv4 } from 'uuid';
+import * as t from 'io-ts';
+import Boom from '@hapi/boom';
+import { termQuery } from '@kbn/observability-plugin/server';
+import type { estypes } from '@elastic/elasticsearch';
+import { getFallbackESUrl } from '../../lib/get_fallback_urls';
+import { createObservabilityOnboardingServerRoute } from '../create_observability_onboarding_server_route';
+import { hasLogMonitoringPrivileges } from '../../lib/api_key/has_log_monitoring_privileges';
+import { createShipperApiKey } from '../../lib/api_key/create_shipper_api_key';
+
+interface CreateKubernetesOnboardingFlowRouteResponse {
+  apiKeyEncoded: string;
+  onboardingId: string;
+  elasticsearchUrl: string;
+}
+
+interface HasKubernetesDataRouteResponse {
+  hasData: boolean;
+}
+
+const createKubernetesOnboardingFlowRoute = createObservabilityOnboardingServerRoute({
+  endpoint: 'POST /internal/observability_onboarding/kubernetes/flow',
+  options: { tags: [] },
+  async handler({
+    context,
+    request,
+    plugins,
+    services,
+  }): Promise<CreateKubernetesOnboardingFlowRouteResponse> {
+    const {
+      elasticsearch: { client },
+    } = await context.core;
+
+    const hasPrivileges = await hasLogMonitoringPrivileges(client.asCurrentUser);
+
+    if (!hasPrivileges) {
+      throw Boom.forbidden(
+        "You don't have enough privileges to start a new onboarding flow. Contact your system administrator to grant you the required privileges."
+      );
+    }
+
+    const fleetStart = await plugins.fleet.start();
+    const packageClient = fleetStart.packageService.asScoped(request);
+
+    await packageClient.ensureInstalledPackage({ pkgName: 'kubernetes' });
+
+    const { encoded: apiKeyEncoded } = await createShipperApiKey(
+      client.asCurrentUser,
+      'onboarding_kubernetes'
+    );
+    const elasticsearchUrlList = plugins.cloud?.setup?.elasticsearchUrl
+      ? [plugins.cloud?.setup?.elasticsearchUrl]
+      : await getFallbackESUrl(services.esLegacyConfigService);
+
+    return {
+      onboardingId: uuidv4(),
+      apiKeyEncoded,
+      elasticsearchUrl: elasticsearchUrlList.length > 0 ? elasticsearchUrlList[0] : '',
+    };
```

Comment:
Hm, good point about telemetry, I missed that. Though I still think it's not worth it to have a saved object in this case as there nothing really to store there besides the ID, and there is no error states that can identify in the flow. I'd rather fire a on-off event when we identify the data was ingested. Wdyt?

Reply:
Sounds good but let's make sure we colocate the code for that with the existing `useFlowProgressTelemetry` so that we don't have that logic in too many different places.